types/types.go

```diff
@@ -256,6 +257,17 @@ type ImageSource interface {
 	// The Digest field is guaranteed to be provided; Size may be -1.
 	// WARNING: The list may contain duplicates, and they are semantically relevant.
 	LayerInfosForCopy(ctx context.Context, instanceDigest *digest.Digest) ([]BlobInfo, error)
+	// GetDeltaManifest returns the delta manifest for the current image, as well as its type, if it exist.
+	// No error is returned if no delta manifest exists, just a nil slice
+	// It may use a remote (= slow) service.
+	// If instanceDigest is not nil, it contains a digest of the specific manifest instance to retrieve deltas for (when the primary manifest is a manifest list);
+	// this never happens if the primary manifest is not a manifest list (e.g. if the source never returns manifest lists).
+	GetDeltaManifest(ctx context.Context, instanceDigest *digest.Digest) ([]byte, string, error)
+	// GetDeltaManifestDestination returns an ImageDestination that can be used to update the delta manifest for this Image.
+	// If deltas are not supported it will return nil
+	// If instanceDigest is not nil, it contains a digest of the specific manifest instance to retrieve deltas for (when the primary manifest is a manifest list);
+	// this never happens if the primary manifest is not a manifest list (e.g. if the source never returns manifest lists).
+	GetDeltaManifestDestination(ctx context.Context, instanceDigest *digest.Digest) (ImageDestination, error)
```

Comment:
This feels intuitively weird. Why should a “source” care about creating destinations? At a first glance this might be a better first for `ImageReference` or maybe `ImageTransport` (or somehow split between “source” and `types.Image`), but I haven’t read the full PR yet.

Reply:
Yes, I think reporting a ImageReference makes more sense. In fact when using this in skopeo i had to do call ImageDestination.Reference() on it, so I already kinda planned on this change.

Reply:
“image”reference… I do need to read the full PR